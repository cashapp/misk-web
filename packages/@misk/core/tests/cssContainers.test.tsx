import React from "react"
import { cleanup, render } from "@testing-library/react"
import {
  CodePreContainer,
  ColumnContainer,
  DesktopWideOnlyContainer,
  FlexContainer,
  MobileNeverContainer,
  MobileOnlyContainer,
  ResponsiveContainer,
  WrapTextContainer,
  ResponsiveAppContainer
} from "../src/cssContainers"

const TestChildren = () => (
  <div>
    <span>{"Test Child"}</span>
  </div>
)

describe("css containers", () => {
  afterEach(cleanup)
  it("CodePreContainer can render", () => {
    const { asFragment } = render(<CodePreContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("CodePreContainer can render with children", () => {
    const { asFragment } = render(
      <CodePreContainer>
        <TestChildren />
      </CodePreContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("ColumnContainer can render", () => {
    const { asFragment } = render(<ColumnContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("ColumnContainer can render with children", () => {
    const { asFragment } = render(
      <ColumnContainer>
        <TestChildren />
      </ColumnContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("DesktopWideOnlyContainer can render", () => {
    const { asFragment } = render(<DesktopWideOnlyContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("DesktopWideOnlyContainer can render with children", () => {
    const { asFragment } = render(
      <DesktopWideOnlyContainer>
        <TestChildren />
      </DesktopWideOnlyContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("FlexContainer can render", () => {
    const { asFragment } = render(<FlexContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("FlexContainer can render with children", () => {
    const { asFragment } = render(
      <FlexContainer>
        <TestChildren />
      </FlexContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("MobileNeverContainer can render", () => {
    const { asFragment } = render(<MobileNeverContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("MobileNeverContainer can render with children", () => {
    const { asFragment } = render(
      <MobileNeverContainer>
        <TestChildren />
      </MobileNeverContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("MobileOnlyContainer can render", () => {
    const { asFragment } = render(<MobileOnlyContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("MobileOnlyContainer can render with children", () => {
    const { asFragment } = render(
      <MobileOnlyContainer>
        <TestChildren />
      </MobileOnlyContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("ResponsiveContainer can render", () => {
    const { asFragment } = render(<ResponsiveContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("ResponsiveContainer can render with children", () => {
    const { asFragment } = render(
      <ResponsiveContainer>
        <TestChildren />
      </ResponsiveContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("ResponsiveAppContainer can render with children", () => {
    const { asFragment } = render(
      <ResponsiveAppContainer>
        <TestChildren />
      </ResponsiveAppContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("WrapTextContainer can render", () => {
    const { asFragment } = render(<WrapTextContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("WrapTextContainer can render with children", () => {
    const { asFragment } = render(
      <WrapTextContainer>
        <TestChildren />
      </WrapTextContainer>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
