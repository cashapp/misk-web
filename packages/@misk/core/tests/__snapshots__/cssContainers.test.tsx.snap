// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css containers CodePreContainer can render 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: Fira Code,Menlo !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  word-wrap: break-word !important;
}

<pre
    class="bp3-code-block emotion-0"
  />
</DocumentFragment>
`;

exports[`css containers CodePreContainer can render with children 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: Fira Code,Menlo !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  word-wrap: break-word !important;
}

<pre
    class="bp3-code-block emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </pre>
</DocumentFragment>
`;

exports[`css containers ColumnContainer can render 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  min-width: 320px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`css containers ColumnContainer can render with children 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  min-width: 320px;
}

<div
    class="emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`css containers DesktopWideOnlyContainer can render 1`] = `
<DocumentFragment>
  @media (max-width:1200px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`css containers DesktopWideOnlyContainer can render with children 1`] = `
<DocumentFragment>
  @media (max-width:1200px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`css containers FlexContainer can render 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-0 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`css containers FlexContainer can render with children 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-0 * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
    class="emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`css containers MobileNeverContainer can render 1`] = `
<DocumentFragment>
  @media (max-width:768px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`css containers MobileNeverContainer can render with children 1`] = `
<DocumentFragment>
  @media (max-width:768px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`css containers MobileOnlyContainer can render 1`] = `
<DocumentFragment>
  @media (min-width:768px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`css containers MobileOnlyContainer can render with children 1`] = `
<DocumentFragment>
  @media (min-width:768px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`css containers ResponsiveAppContainer can render with children 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  top: 110px;
  padding-left: 5px;
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    width: 1170px;
  }
}

<div
    class="emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`css containers ResponsiveContainer can render 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    width: 1170px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`css containers ResponsiveContainer can render with children 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 auto;
}

@media (min-width:768px) {
  .emotion-0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    width: 1170px;
  }
}

<div
    class="emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`css containers WrapTextContainer can render 1`] = `
<DocumentFragment>
  .emotion-0 {
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  word-wrap: break-word !important;
}

<span
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`css containers WrapTextContainer can render with children 1`] = `
<DocumentFragment>
  .emotion-0 {
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  word-wrap: break-word !important;
}

<span
    class="emotion-0"
  >
    <div>
      <span>
        Test Child
      </span>
    </div>
  </span>
</DocumentFragment>
`;
